app/controllers/contacts_controller.rb

```diff
@@ -1,8 +1,14 @@
 class ContactsController < ApplicationController
   before_action :set_contact, only: %i[ edit update destroy ]
+  before_action :set_contact_types, only: %i[ create edit new update ]
 
   def index
     @contacts = Contact.all
+    @contacts.each do |contact|
+      contact.contact_type_display_value = CategoryReference.display_value_by_category_and_code(
+        :contact_type, contact.contact_type
+      )
+    end
```

Comment:
I have my concern about this approach to get the display name of contact type for each contact because the whole address book will be loaded, it may cause issue when the address book gets big,  is it possible to create relationship (join) to get this value, so only first page will be linked up to the reference table.     For this relationship, I think it won't cause trouble for creating a new contact type or a new contact, ?